ovirt-image-template: add image_cache_download option

tremble opened the pull request:
When dealing with local images you may not want to cache the downloaded
file.  Setting this option to false will ensure that the image is
deleted (before and after running the role)

The default behaviour right now can be confusing because /tmp/ovirt_image_data can be left lying around by previous runs of the role

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for an oVirt project! 

I am the bot who triggers CI builds on the oVirt Jenkins server. 
In order to allow this PR to run fully automated tests, you can do one of the following options:

1. Asking one of the project maintainers to add you to the project team ( it's done via GitHub admin page for the project )
2. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the oVirt organization, so you'll be able to run tests for all oVirt's projects.
3. Asking one of the maintainers to run the 'ci test please' on the PR.


Note: You might see multiple messages like this showing up on a single PR, this is due to a known bug in the GitHub trigger plugin on Jenkins, the oVirt infra team is working on fixing this via new testing flow in the near future, so we'll be able to post a single message per PR.

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for an oVirt project! 

I am the bot who triggers CI builds on the oVirt Jenkins server. 
In order to allow this PR to run fully automated tests, you can do one of the following options:

1. Asking one of the project maintainers to add you to the project team ( it's done via GitHub admin page for the project )
2. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the oVirt organization, so you'll be able to run tests for all oVirt's projects.
3. Asking one of the maintainers to run the 'ci test please' on the PR.


Note: You might see multiple messages like this showing up on a single PR, this is due to a known bug in the GitHub trigger plugin on Jenkins, the oVirt infra team is working on fixing this via new testing flow in the near future, so we'll be able to post a single message per PR.

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for an oVirt project! 

I am the bot who triggers CI builds on the oVirt Jenkins server. 
In order to allow this PR to run fully automated tests, you can do one of the following options:

1. Asking one of the project maintainers to add you to the project team ( it's done via GitHub admin page for the project )
2. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the oVirt organization, so you'll be able to run tests for all oVirt's projects.
3. Asking one of the maintainers to run the 'ci test please' on the PR.


Note: You might see multiple messages like this showing up on a single PR, this is due to a known bug in the GitHub trigger plugin on Jenkins, the oVirt infra team is working on fixing this via new testing flow in the near future, so we'll be able to post a single message per PR.

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for an oVirt project! 

I am the bot who triggers CI builds on the oVirt Jenkins server. 
In order to allow this PR to run fully automated tests, you can do one of the following options:

1. Asking one of the project maintainers to add you to the project team ( it's done via GitHub admin page for the project )
2. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the oVirt organization, so you'll be able to run tests for all oVirt's projects.
3. Asking one of the maintainers to run the 'ci test please' on the PR.


Note: You might see multiple messages like this showing up on a single PR, this is due to a known bug in the GitHub trigger plugin on Jenkins, the oVirt infra team is working on fixing this via new testing flow in the near future, so we'll be able to post a single message per PR.

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for an oVirt project! 

I am the bot who triggers CI builds on the oVirt Jenkins server. 
In order to allow this PR to run fully automated tests, you can do one of the following options:

1. Asking one of the project maintainers to add you to the project team ( it's done via GitHub admin page for the project )
2. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the oVirt organization, so you'll be able to run tests for all oVirt's projects.
3. Asking one of the maintainers to run the 'ci test please' on the PR.


Note: You might see multiple messages like this showing up on a single PR, this is due to a known bug in the GitHub trigger plugin on Jenkins, the oVirt infra team is working on fixing this via new testing flow in the near future, so we'll be able to post a single message per PR.

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for an oVirt project! 

I am the bot who triggers CI builds on the oVirt Jenkins server. 
In order to allow this PR to run fully automated tests, you can do one of the following options:

1. Asking one of the project maintainers to add you to the project team ( it's done via GitHub admin page for the project )
2. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the oVirt organization, so you'll be able to run tests for all oVirt's projects.
3. Asking one of the maintainers to run the 'ci test please' on the PR.


Note: You might see multiple messages like this showing up on a single PR, this is due to a known bug in the GitHub trigger plugin on Jenkins, the oVirt infra team is working on fixing this via new testing flow in the near future, so we'll be able to post a single message per PR.

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for an oVirt project! 

I am the bot who triggers CI builds on the oVirt Jenkins server. 
In order to allow this PR to run fully automated tests, you can do one of the following options:

1. Asking one of the project maintainers to add you to the project team ( it's done via GitHub admin page for the project )
2. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the oVirt organization, so you'll be able to run tests for all oVirt's projects.
3. Asking one of the maintainers to run the 'ci test please' on the PR.


Note: You might see multiple messages like this showing up on a single PR, this is due to a known bug in the GitHub trigger plugin on Jenkins, the oVirt infra team is working on fixing this via new testing flow in the near future, so we'll be able to post a single message per PR.

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for an oVirt project! 

I am the bot who triggers CI builds on the oVirt Jenkins server. 
In order to allow this PR to run fully automated tests, you can do one of the following options:

1. Asking one of the project maintainers to add you to the project team ( it's done via GitHub admin page for the project )
2. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the oVirt organization, so you'll be able to run tests for all oVirt's projects.
3. Asking one of the maintainers to run the 'ci test please' on the PR.


Note: You might see multiple messages like this showing up on a single PR, this is due to a known bug in the GitHub trigger plugin on Jenkins, the oVirt infra team is working on fixing this via new testing flow in the near future, so we'll be able to post a single message per PR.

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for an oVirt project! 

I am the bot who triggers CI builds on the oVirt Jenkins server. 
In order to allow this PR to run fully automated tests, you can do one of the following options:

1. Asking one of the project maintainers to add you to the project team ( it's done via GitHub admin page for the project )
2. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the oVirt organization, so you'll be able to run tests for all oVirt's projects.
3. Asking one of the maintainers to run the 'ci test please' on the PR.


Note: You might see multiple messages like this showing up on a single PR, this is due to a known bug in the GitHub trigger plugin on Jenkins, the oVirt infra team is working on fixing this via new testing flow in the near future, so we'll be able to post a single message per PR.

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for an oVirt project! 

I am the bot who triggers CI builds on the oVirt Jenkins server. 
In order to allow this PR to run fully automated tests, you can do one of the following options:

1. Asking one of the project maintainers to add you to the project team ( it's done via GitHub admin page for the project )
2. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the oVirt organization, so you'll be able to run tests for all oVirt's projects.
3. Asking one of the maintainers to run the 'ci test please' on the PR.


Note: You might see multiple messages like this showing up on a single PR, this is due to a known bug in the GitHub trigger plugin on Jenkins, the oVirt infra team is working on fixing this via new testing flow in the near future, so we'll be able to post a single message per PR.

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for an oVirt project! 

I am the bot who triggers CI builds on the oVirt Jenkins server. 
In order to allow this PR to run fully automated tests, you can do one of the following options:

1. Asking one of the project maintainers to add you to the project team ( it's done via GitHub admin page for the project )
2. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the oVirt organization, so you'll be able to run tests for all oVirt's projects.
3. Asking one of the maintainers to run the 'ci test please' on the PR.


Note: You might see multiple messages like this showing up on a single PR, this is due to a known bug in the GitHub trigger plugin on Jenkins, the oVirt infra team is working on fixing this via new testing flow in the near future, so we'll be able to post a single message per PR.

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for an oVirt project! 

I am the bot who triggers CI builds on the oVirt Jenkins server. 
In order to allow this PR to run fully automated tests, you can do one of the following options:

1. Asking one of the project maintainers to add you to the project team ( it's done via GitHub admin page for the project )
2. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the oVirt organization, so you'll be able to run tests for all oVirt's projects.
3. Asking one of the maintainers to run the 'ci test please' on the PR.


Note: You might see multiple messages like this showing up on a single PR, this is due to a known bug in the GitHub trigger plugin on Jenkins, the oVirt infra team is working on fixing this via new testing flow in the near future, so we'll be able to post a single message per PR.

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for an oVirt project! 

I am the bot who triggers CI builds on the oVirt Jenkins server. 
In order to allow this PR to run fully automated tests, you can do one of the following options:

1. Asking one of the project maintainers to add you to the project team ( it's done via GitHub admin page for the project )
2. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the oVirt organization, so you'll be able to run tests for all oVirt's projects.
3. Asking one of the maintainers to run the 'ci test please' on the PR.


Note: You might see multiple messages like this showing up on a single PR, this is due to a known bug in the GitHub trigger plugin on Jenkins, the oVirt infra team is working on fixing this via new testing flow in the near future, so we'll be able to post a single message per PR.

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for an oVirt project! 

I am the bot who triggers CI builds on the oVirt Jenkins server. 
In order to allow this PR to run fully automated tests, you can do one of the following options:

1. Asking one of the project maintainers to add you to the project team ( it's done via GitHub admin page for the project )
2. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the oVirt organization, so you'll be able to run tests for all oVirt's projects.
3. Asking one of the maintainers to run the 'ci test please' on the PR.


Note: You might see multiple messages like this showing up on a single PR, this is due to a known bug in the GitHub trigger plugin on Jenkins, the oVirt infra team is working on fixing this via new testing flow in the near future, so we'll be able to post a single message per PR.

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for an oVirt project! 

I am the bot who triggers CI builds on the oVirt Jenkins server. 
In order to allow this PR to run fully automated tests, you can do one of the following options:

1. Asking one of the project maintainers to add you to the project team ( it's done via GitHub admin page for the project )
2. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the oVirt organization, so you'll be able to run tests for all oVirt's projects.
3. Asking one of the maintainers to run the 'ci test please' on the PR.


Note: You might see multiple messages like this showing up on a single PR, this is due to a known bug in the GitHub trigger plugin on Jenkins, the oVirt infra team is working on fixing this via new testing flow in the near future, so we'll be able to post a single message per PR.

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for an oVirt project! 

I am the bot who triggers CI builds on the oVirt Jenkins server. 
In order to allow this PR to run fully automated tests, you can do one of the following options:

1. Asking one of the project maintainers to add you to the project team ( it's done via GitHub admin page for the project )
2. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the oVirt organization, so you'll be able to run tests for all oVirt's projects.
3. Asking one of the maintainers to run the 'ci test please' on the PR.


Note: You might see multiple messages like this showing up on a single PR, this is due to a known bug in the GitHub trigger plugin on Jenkins, the oVirt infra team is working on fixing this via new testing flow in the near future, so we'll be able to post a single message per PR.

machacekondra commented:
add to whitelis

machacekondra commented:
add to whitelist

machacekondra commented on roles/ovirt-image-template/tasks/qcow2_image.yml:
```diff
@@ -1,4 +1,10 @@
+
 - name: Download the qcow image
   get_url:
     url: "{{ qcow_url }}"
```
What about:

```yaml
get_url:
  force: "{{ not image_cache_download }}"
  ...
```

And remove the `file` task?

machacekondra commented on roles/ovirt-image-template/tasks/qcow2_image.yml:
```diff
@@ -101,6 +108,12 @@
+      file:
+        path: "{{ image_path }}"
+        state: absent
+      when:  image_cache_download == false
```
Any reason to use `== false` instead of `not`

tremble replied:
@machacekondra Made the recommended changes

tremble committed:
ovirt-image-template: add image_cache_download option

When dealing with local images you may not want to cache the downloaded
file. Setting this option to false will force a re-download and ensure
that the image is deleted after running the role.

The default behaviour right now can be confusing because /tmp/ovirt_image_data can be left lying around by previous runs of the role

machacekondra reviewed (approved):
Thanks!
